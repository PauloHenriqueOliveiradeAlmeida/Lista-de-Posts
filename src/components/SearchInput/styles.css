.search-input {
    outline: none;
    padding: 0.5rem 1rem;
    border-radius: 0.2rem;
    border: 1px solid black;
}