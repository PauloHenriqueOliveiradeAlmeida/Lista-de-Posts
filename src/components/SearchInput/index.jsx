import "./styles.css";

function SearchInput({handleChange, searchValue}) {
    return (
        <input
        className="search-input"
        type="search"
        placeholder="Pesquisar..."
        value={searchValue}
        onChange={handleChange} />
    )
}

export default SearchInput;