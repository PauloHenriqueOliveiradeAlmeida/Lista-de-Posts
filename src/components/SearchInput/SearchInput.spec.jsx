import userEvent from "@testing-library/user-event";
import SearchInput from ".";
import { render, screen } from "@testing-library/react";

describe("<SearchInput />", () => {
    it("should have a value of a searchValue", () => {
        const fn = jest.fn();
        render(<SearchInput handleChange={fn} searchValue="value"/>);

        expect(screen.getByPlaceholderText(/Pesquisar.../i).value).toBe("value")

    });

    it("should call handleChange function where change SearchInput value", () => {
        const fn = jest.fn();
        render(<SearchInput handleChange={fn} searchValue="value"/>);
        userEvent.type(screen.getByPlaceholderText(/Pesquisar.../i), "value")

        expect(fn).toHaveBeenCalledTimes("value".length)
    });

    it("should match snapshot", () => {
        const fn = jest.fn();
        const { container } = render(<SearchInput handleChange={fn} searchValue="value"/>);

        expect(container).toMatchSnapshot();
    });
});
