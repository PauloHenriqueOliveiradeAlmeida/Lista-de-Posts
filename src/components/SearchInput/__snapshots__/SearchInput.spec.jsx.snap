// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should match snapshot 1`] = `
<div>
  <input
    class="search-input"
    placeholder="Pesquisar..."
    type="search"
    value="value"
  />
</div>
`;
